applications/StructuralMechanicsApplication/custom_elements/small_displacement_mixed_volumetric_strain_element.cpp

```diff
@@ -943,14 +943,23 @@ void SmallDisplacementMixedVolumetricStrainElement::CalculateAnisotropyTensor(co
     // The k_iso and mu_iso coefficients are obtained from the Frobenius norm minimization of the closest isotropic tensor to C
     // Such minimization is constrained such that the compressibility of the new C_iso is equivalent to the input anisotropic C one
     Vector u = ZeroVector(strain_size);
-    for (IndexType i = 0; i < dim; ++i) {
-        u(i) = 1.0 / std::sqrt(dim);
-    }
-    const Matrix J = outer_prod(u, u);
     Matrix I = ZeroMatrix(strain_size);
-    for (IndexType i = 0; i < strain_size; ++i){
-        I(i, i) = i < dim ? 1.0 : 0.5;
+    if (strain_size == 4) {
```

Comment:
I'm very concerned about this... As explained in the comment above these coefficients come from a delicate minimization operation.

I've checked the current cpp and Python tests and there is no direct coverage for these operations. Having said this, we must add an extra test before merging this changes. I will take care ASAP.